srcpkgs/python3/patches/configure-ppc-musl.patch

```diff
@@ -0,0 +1,14 @@
+--- configure.orig	2018-06-27 01:39:50.000000000 +0200
++++ configure	2018-08-25 08:20:56.420702095 +0200
+@@ -5286,7 +5286,11 @@
+         powerpc64-linux-gnu
+ #  endif
+ # elif defined(__powerpc__)
++#  if definded(__GLIBC__)
+         powerpc-linux-gnu
++#  else
++        powerpc-linux-musl
```

Comment:
why is this patch necessary? none of the musl packages we have already define a python platform triplet; they all use the default GNU triplet and it doesn't seem to harm anything. And for example Adelie Linux, a distro that purely uses musl and has support for powerpc, doesn't patch the triplet either.

Reply:
Without it, python doesn't recognise the powerpc-linux-musl triplet; says it's not a valid target. I'm not aware of how Adelie Linux does it, perhaps they do other things differently too.

Reply:
to demonstrate, here is what i get when i unpack the python3 package for armv6l-musl

```
$ find . -name '*.so'|xargs file       
./usr/lib/libpython3.so:                                                                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/libpython3.6m.so:                                                             symbolic link to libpython3.6m.so.1.0
./usr/lib/python3.6/lib-dynload/_sha512.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_multibytecodec.cpython-36m-arm-linux-gnueabihf.so:     ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/readline.cpython-36m-arm-linux-gnueabihf.so:            ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_md5.cpython-36m-arm-linux-gnueabihf.so:                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_struct.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_sha1.cpython-36m-arm-linux-gnueabihf.so:               ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/ossaudiodev.cpython-36m-arm-linux-gnueabihf.so:         ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_curses_panel.cpython-36m-arm-linux-gnueabihf.so:       ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/parser.cpython-36m-arm-linux-gnueabihf.so:              ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/mmap.cpython-36m-arm-linux-gnueabihf.so:                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/select.cpython-36m-arm-linux-gnueabihf.so:              ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_testcapi.cpython-36m-arm-linux-gnueabihf.so:           ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/math.cpython-36m-arm-linux-gnueabihf.so:                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/unicodedata.cpython-36m-arm-linux-gnueabihf.so:         ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_asyncio.cpython-36m-arm-linux-gnueabihf.so:            ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_testimportmultiple.cpython-36m-arm-linux-gnueabihf.so: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_pickle.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_curses.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_socket.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_csv.cpython-36m-arm-linux-gnueabihf.so:                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_lzma.cpython-36m-arm-linux-gnueabihf.so:               ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/grp.cpython-36m-arm-linux-gnueabihf.so:                 ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_codecs_jp.cpython-36m-arm-linux-gnueabihf.so:          ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_multiprocessing.cpython-36m-arm-linux-gnueabihf.so:    ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_random.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/cmath.cpython-36m-arm-linux-gnueabihf.so:               ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_ctypes_test.cpython-36m-arm-linux-gnueabihf.so:        ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_sqlite3.cpython-36m-arm-linux-gnueabihf.so:            ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_codecs_iso2022.cpython-36m-arm-linux-gnueabihf.so:     ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_datetime.cpython-36m-arm-linux-gnueabihf.so:           ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/spwd.cpython-36m-arm-linux-gnueabihf.so:                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_testmultiphase.cpython-36m-arm-linux-gnueabihf.so:     ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_blake2.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_bisect.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_sha256.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/zlib.cpython-36m-arm-linux-gnueabihf.so:                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_codecs_cn.cpython-36m-arm-linux-gnueabihf.so:          ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_dbm.cpython-36m-arm-linux-gnueabihf.so:                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/termios.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_decimal.cpython-36m-arm-linux-gnueabihf.so:            ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_heapq.cpython-36m-arm-linux-gnueabihf.so:              ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_opcode.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/fcntl.cpython-36m-arm-linux-gnueabihf.so:               ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_testbuffer.cpython-36m-arm-linux-gnueabihf.so:         ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_posixsubprocess.cpython-36m-arm-linux-gnueabihf.so:    ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_codecs_tw.cpython-36m-arm-linux-gnueabihf.so:          ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/array.cpython-36m-arm-linux-gnueabihf.so:               ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_json.cpython-36m-arm-linux-gnueabihf.so:               ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/syslog.cpython-36m-arm-linux-gnueabihf.so:              ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_codecs_hk.cpython-36m-arm-linux-gnueabihf.so:          ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_bz2.cpython-36m-arm-linux-gnueabihf.so:                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_ssl.cpython-36m-arm-linux-gnueabihf.so:                ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_elementtree.cpython-36m-arm-linux-gnueabihf.so:        ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_codecs_kr.cpython-36m-arm-linux-gnueabihf.so:          ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_hashlib.cpython-36m-arm-linux-gnueabihf.so:            ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/pyexpat.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/xxlimited.cpython-36m-arm-linux-gnueabihf.so:           ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/resource.cpython-36m-arm-linux-gnueabihf.so:            ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/audioop.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_lsprof.cpython-36m-arm-linux-gnueabihf.so:             ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/binascii.cpython-36m-arm-linux-gnueabihf.so:            ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_crypt.cpython-36m-arm-linux-gnueabihf.so:              ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_gdbm.cpython-36m-arm-linux-gnueabihf.so:               ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
./usr/lib/python3.6/lib-dynload/_sha3.cpython-36m-arm-linux-gnueabihf.so:               ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, stripped
```

Reply:
I don't know. It doesn't pass the configuration without it on ppc-musl.

Reply:
This is something that should be investigated, as I don't see why it's necessary here but not on other platforms.

Reply:
any news on this? I would like to see ppc support merged and this seems to be the only issue left.

Reply:
Let's maybe merge the `common/` files and cross toolchain first, and give the other stuff a more detailed review. For example, I also have some concerns about the linux kernel support, it doesn't seem to correctly handle header installation for the powerpc arch, it doesn't seem to correctly handle the `zImage` vs `vmlinux` stuff (unless things are different on 32-bit, it should be using `vmlinux` directly like 64-bit), maybe some other things; it will also need rebasing, as conflicts have been introduced in the meantime.

Reply:
Also, if we can have a cross-toolchain support, maybe other people can help address the concerns about the other packages because they will be able to actually build them.

Reply:
Sounds good. @stenstorp could you please create smaller PRs so we can get the minimal basics reviewed and merged first?

Reply:
let's start with the minimum changes that are necessary to add the architecture and cross-compile a simple package that is not restricted to a specific architecture.

Reply:
Done, only the stuff in common, cross-vpkg-dummy and cross-powerpc-*.